### DISCUSSION TITLE
/proc/spl/kstat/zfs/arcstats documentation?

### DISCUSSION THREAD

**Discussion by joakimlemb** [openzfs/zfs#12250] [General]
Is there any documentation that describes all the parameters and values for /proc/spl/kstat/zfs/arcstats?

I found this: https://github.com/influxdata/telegraf/tree/master/plugins/inputs/zfs
But that only has described about 10% of the parameters, and does not explain what type the value is (byte, bits, blocks, count, etc...).

For example:
evict_l2_eligible 4 1655942671872

The telegraf readme states "We evicted something from the arc, and it’s not in the l2 this is sad. (maybe we hadn’t had enough time to store it there)". 

But is the value bytes or ashift blocks or maybe in 512-byte sectors like LBA written in S.M.A.R.T?

**Comment by richardelling** [comment]
The value is in bytes.

The documentation in this respect is lacking. This somewhat exacerbated by collectors like telegraf
which grabs all of the ARC kstats available. While some are fairly obvious to folks who work with caches
regularly, there isn't much practical application. In a few cases, I've tried to do this in the analysis tools
themselves, such as grafana dashboards like
https://github.com/richardelling/grafana-dashboards/blob/master/linux/Compressed-ARC-example.png
Alas, I don't have a complete set of grafana dashboards publicly available. There are a few in the grafana
public dashboard repo, but they are mostly graphs of data without description of what the graphs might
convey.
